libretroshare/src/libretroshare.pro

```diff
@@ -861,45 +861,69 @@ rs_jsonapi {
+                git submodule update --init --recommend-shallow dependency/catch && \
+                git submodule update --init --recommend-shallow dependency/kashmir && \
+                mkdir -p $${RESTBED_BUILD_PATH}; cd $${RESTBED_BUILD_PATH} && \
+                cmake -DCMAKE_CXX_COMPILER=$$QMAKE_CXX -G \"MSYS Makefiles\" -DBUILD_SSL=OFF \
+                    -DCMAKE_INSTALL_PREFIX=. -B. -H$$shell_path($${RESTBED_SRC_PATH}) && \
+                make
+        } else {
+            genrestbedlib.commands = \
+                cd $${RS_SRC_PATH};\
+                git submodule update --init --recommend-shallow supportlibs/restbed;\
+                cd $${RESTBED_SRC_PATH};\
+                git submodule update --init --recommend-shallow dependency/asio;\
+                git submodule update --init --recommend-shallow dependency/catch;\
+                git submodule update --init --recommend-shallow dependency/kashmir;\
+                mkdir -p $${RESTBED_BUILD_PATH}; cd $${RESTBED_BUILD_PATH};\
+                cmake -DCMAKE_CXX_COMPILER=$$QMAKE_CXX -DBUILD_SSL=OFF \
+                    -DCMAKE_INSTALL_PREFIX=. -B. -H$$shell_path($${RESTBED_SRC_PATH});\
+                make
+        }
+        QMAKE_EXTRA_COMPILERS += genrestbedlib
+
+        RESTBED_HEADER_FILE=$$clean_path($${RESTBED_BUILD_PATH}/include/restbed)
+        genrestbedheader.name = Generating restbed header.
+        genrestbedheader.input = genrestbedlib.output
+        genrestbedheader.output = $${RESTBED_HEADER_FILE}
+        genrestbedheader.CONFIG += target_predeps combine no_link
+        genrestbedheader.variable_out = HEADERS
+        genrestbedheader.commands = cd $${RESTBED_BUILD_PATH} && make install
+        QMAKE_EXTRA_COMPILERS += genrestbedheader
     }
 
-    PRE_TARGETDEPS *= $${JSONAPI_GENERATOR_EXE}
     INCLUDEPATH *= $${JSONAPI_GENERATOR_OUT}
-    GENERATED_HEADERS += $${WRAPPERS_INCL_FILE}
-
-    jsonwrappersincl.target = $${WRAPPERS_INCL_FILE}
-    jsonwrappersincl.commands = \
+    DEPENDPATH *= $${JSONAPI_GENERATOR_OUT}
+    APIHEADERS = $$files($${RS_SRC_PATH}/libretroshare/src/retroshare/*.h)
+    #Make sure that the jsonapigenerator executable are ready
+    APIHEADERS += $${JSONAPI_GENERATOR_EXE}
+
+    genjsonapi.name = Generating jsonapi headers.
+    genjsonapi.input = APIHEADERS
+    genjsonapi.output = $${WRAPPERS_INCL_FILE} $${WRAPPERS_REG_FILE}
+    genjsonapi.clean = $${WRAPPERS_INCL_FILE} $${WRAPPERS_REG_FILE}
+    genjsonapi.CONFIG += target_predeps combine no_link
+    genjsonapi.variable_out = HEADERS
+    genjsonapi.commands = \
         mkdir -p $${JSONAPI_GENERATOR_OUT} && \
         cp $${DOXIGEN_CONFIG_SRC} $${DOXIGEN_CONFIG_OUT} && \
-        echo OUTPUT_DIRECTORY=$$shell_path($${JSONAPI_GENERATOR_OUT}) >> $${DOXIGEN_CONFIG_OUT} && \
-        echo INPUT=$$shell_path($${DOXIGEN_INPUT_DIRECTORY}) >> $${DOXIGEN_CONFIG_OUT} && \
+        echo OUTPUT_DIRECTORY=$${JSONAPI_GENERATOR_OUT} >> $${DOXIGEN_CONFIG_OUT} && \
```

Comment:
Are you sure this works better without the `$$shell_path(...)` ?

Reply:
With shell_path qmake will print Linux like paths into the doxygen config, and doxygen will throw an error.